test/cql-pytest/test_describe.py

```diff
@@ -733,6 +733,36 @@ def test_table_options_quoting(cql, test_keyspace):
     finally:
         cql.execute(f"DROP TYPE {test_keyspace}.\"{type_name}\"")
 
+@pytest.mark.parametrize("test_keyspace",
+                         [pytest.param("tablets", marks=[pytest.mark.xfail(reason="issue #16317")]), "vnodes"],
+                         indirect=True)
+def test_hide_cdc_table(scylla_only, cql, test_keyspace):
+    cdc_table_suffix = f"_scylla_cdc_log"
+    
+    with new_test_table(cql, test_keyspace, "a int primary key, b int", "WITH cdc = {'enabled': true}") as t:
+        t_name = t.split('.')[1]
+
+        # Check 't_scylla_cdc_log' is hidden in DESC TABLES
+        desc_tables = cql.execute("DESC TABLES")
+        assert t_name + cdc_table_suffix not in [r.name for r in desc_tables]
+
+        # Check 't_scylla_cdc_log' is hidden in DESC KEYSPACE ks
+        desc_keyspace = cql.execute(f"DESC KEYSPACE {test_keyspace}").all()
+        assert t_name + cdc_table_suffix not in [r.name for r in desc_keyspace]
+
+        # Check 't_scylla_cdc_log' is hidden in DESC SCHEMA
+        desc_schema = cql.execute("DESC SCHEMA")
+        assert t_name + cdc_table_suffix not in [r.name for r in desc_schema]
+
+        # Check 't_scylla_cdc_log' can be described directly
+        desc_cdc_table = cql.execute(f"DESC TABLE {t + cdc_table_suffix}")
+        assert f"CREATE TABLE {t + cdc_table_suffix}" in desc_cdc_table.one().create_statement
+
+        # Drop keyspace and try to recreate it with cdc table
+        cql.execute(f"DROP KEYSPACE {test_keyspace}")
+        for row in desc_keyspace:
+            cql.execute(row.create_statement)
```

Comment:
I think this last part (deleting the keyspace and recreating it) is wrong and unnecessary:

1. I think it's unnecessary because I don't see how it is it related to or relevant to this test?
2. It is wrong because test_keyspace is shared by many tests and other fixtures might have created tables in it, perhaps even tables with known content - and this "test_keyspace" will delete all these tables!

If you really want to test the round-trip of DESK KEYSPACE and CREATE (again, I don't understand why you'd want to do this in this test), you should do it on a new keyspace created just for your test - not on the global test_keyspace.